src/librustc_middle/ty/print/pretty.rs

```diff
@@ -689,9 +687,7 @@ pub trait PrettyPrinter<'tcx>:
                         }
                     }
                 } else {
-                    // Cross-crate closure types should only be
-                    // visible in codegen bug reports, I imagine.
-                    p!(write("@{:?}", did));
+                    p!(write("@{}", self.tcx().def_path_str(did)));
```

Comment:
You can see above that `p!(write("@"), print_def_path(did, substs));` is the right way to do this (used in the local case with `span_free_formats`).

Reply:
Oh, overlooked this comment, sorry! Will address it soonish.